tests/unit/test_repo.py

```diff
@@ -1,13 +1,76 @@
-import os
+from dvc.repo import Repo
+from dvc.scm.git import GitTree
+from dvc.system import System
+from dvc.utils.compat import fspath
 
-from tests.basic_env import TestDvc
 
+def test_is_dvc_internal(tmp_dir, dvc):
+    assert dvc.is_dvc_internal(fspath(tmp_dir / ".dvc" / "file"))
+    assert not dvc.is_dvc_internal("file")
 
-class TestIsDvcInternal(TestDvc):
-    def test_return_false_on_non_dvc_file(self):
-        path = os.path.join("path", "to-non-.dvc", "file")
-        self.assertFalse(self.dvc.is_dvc_internal(path))
 
-    def test_return_true_on_dvc_internal_file(self):
-        path = os.path.join("path", "to", ".dvc", "file")
-        self.assertTrue(path)
+def test_destroy(tmp_dir, dvc):
+    dvc.config.set("cache", "type", "symlink")
+
+    tmp_dir.dvc_gen("file", "text")
+    tmp_dir.dvc_gen({"dir": {"file": "lorem", "subdir/file": "ipsum"}})
+
+    dvc.destroy()
+
+    # Remove all the files related to DVC
+    assert not (tmp_dir / ".dvc").exists()
+    assert not (tmp_dir / "file.dvc").exists()
+    assert not (tmp_dir / "dir.dvc").exists()
+
+    # Leave the rest of the files
+    assert (tmp_dir / "file").is_file()
+    assert (tmp_dir / "dir").is_dir()
+    assert (tmp_dir / "dir" / "file").is_file()
+    assert (tmp_dir / "dir" / "subdir" / "file").is_file()
+
+    # Make sure that data was unprotected after `destroy`
+    assert not System.is_symlink(fspath(tmp_dir / "foo"))
+    assert not System.is_symlink(fspath(tmp_dir / "file"))
+    assert not System.is_symlink(fspath(tmp_dir / "dir"))
+    assert not System.is_symlink(fspath(tmp_dir / "dir" / "file"))
+    assert not System.is_symlink(fspath(tmp_dir / "dir" / "subdir" / "file"))
+
+
+def test_collect(tmp_dir, scm, dvc, run_copy):
```

Comment:
didn't get why we are creating branches, why not doing the following:

```python
def test_collect(tmp_dir, dvc, run_copy):
    def collect_outs(*args, **kwargs):
        return [
            str(out.path_info)
            for stage in dvc.collect(*args, **kwargs)
            for out in stage.outs
        ]

    tmp_dir.dvc_gen("foo", "foo")
    run_copy("foo", "bar")

    assert ["foo", "bar"] == collect_outs("bar.dvc", with_deps=True)

    run_copy("bar", "buzz")

    assert ["foo", "bar", "buzz"] == collect_outs("buzz.dvc", with_deps=True)
    assert ["buzz"] == collect_outs("buzz.dvc", with_deps=False)
```

Reply:
No idea) May look at blame and ask whoever added this. But, again, this migration is not about unittests, so I suggest just revert all unittest changes for now. Or we might stuck with this, which is not good.

Reply:
@efiop , could you shade some light?